framework/functions.php

```diff
@@ -257,18 +257,26 @@ function fw_menu_output ( $menu, $level, $type, $classes ) {
 	echo '">';
 
 	foreach ( $menu as $item ) {
-	
+		$is_item_of_current_page = false;
+
 		echo '<li class="';
 
-		if (
-			isset ( $GLOBALS['vars']['current_url'] ) &&
-			$GLOBALS['vars']['current_url'] == $item['url']
-		) {
-			echo 'current-nav-item ';
+		if ( isset ( $GLOBALS['vars']['current_url'] ) ) {
+			$current_host = parse_url( $GLOBALS['vars']['current_url'], PHP_URL_HOST );
+			$current_path = parse_url( $GLOBALS['vars']['current_url'], PHP_URL_PATH );
+			$current_url = $current_host . $current_path;
+
+			$item_host = parse_url( $item['url'], PHP_URL_HOST );
+			$item_path = parse_url( $item['url'], PHP_URL_PATH );
+			$item_url =  $item_host . $item_path;
+
+			if ( $current_url == $item_url ) {
+				$is_item_of_current_page = true;
+				echo 'current-nav-item ' . $current_url . ' ';
+			}
 		}
 
 		// if the page is an ancestor of the current ID
-
 		if ( 
 			isset ( $GLOBALS['vars']['current_ancestors'] ) &&
 			in_array ( $item['id'], $GLOBALS['vars']['current_ancestors'] )
```

Comment:
This `if` doesn't seem to work as expected: it doesn't highlight the menu item when we are in a sub page.

Note: I don't mind if we don't use `$GLOBALS['vars']['current_ancestors']`, we can instead directly use `get_post_ancestors()`.

To reproduce:
* In the hamburger menu, click on "About"
* In the About page, click on any sub-page in the menu (ex: click on "Our Data")
* Open the hamburger menu

Expected:
* The "About" menu is highlighted

Actual:
* The "About" menu is not highlighted